--- a/net/minecraft/block/CocoaBlock.java
+++ b/net/minecraft/block/CocoaBlock.java
@@ -17,7 +17,7 @@
 import net.minecraft.world.World;
 import net.minecraft.world.server.ServerWorld;
 
-public class CocoaBlock extends HorizontalBlock implements IGrowable {
+public class CocoaBlock extends HorizontalBlock implements IGrowable, net.minecraftforge.common.IPlantable {
    public static final IntegerProperty field_176501_a = BlockStateProperties.field_208167_T;
    protected static final VoxelShape[] field_185535_b = new VoxelShape[]{Block.func_208617_a(11.0D, 7.0D, 6.0D, 15.0D, 12.0D, 10.0D), Block.func_208617_a(9.0D, 5.0D, 5.0D, 15.0D, 12.0D, 11.0D), Block.func_208617_a(7.0D, 3.0D, 4.0D, 15.0D, 12.0D, 12.0D)};
    protected static final VoxelShape[] field_185536_c = new VoxelShape[]{Block.func_208617_a(1.0D, 7.0D, 6.0D, 5.0D, 12.0D, 10.0D), Block.func_208617_a(1.0D, 5.0D, 5.0D, 7.0D, 12.0D, 11.0D), Block.func_208617_a(1.0D, 3.0D, 4.0D, 9.0D, 12.0D, 12.0D)};
@@ -30,10 +30,11 @@
    }
 
    public void func_225534_a_(BlockState p_225534_1_, ServerWorld p_225534_2_, BlockPos p_225534_3_, Random p_225534_4_) {
-      if (p_225534_2_.field_73012_v.nextInt(5) == 0) {
+      if (true) {
          int i = p_225534_1_.func_177229_b(field_176501_a);
-         if (i < 2) {
+         if (i < 2 && net.minecraftforge.common.ForgeHooks.onCropsGrowPre(p_225534_2_, p_225534_3_, p_225534_1_, p_225534_2_.field_73012_v.nextInt(5) == 0)) {
             p_225534_2_.func_180501_a(p_225534_3_, p_225534_1_.func_206870_a(field_176501_a, Integer.valueOf(i + 1)), 2);
+            net.minecraftforge.common.ForgeHooks.onCropsGrowPost(p_225534_2_, p_225534_3_, p_225534_1_);
          }
       }
 
@@ -81,7 +82,7 @@
       return p_196271_2_ == p_196271_1_.func_177229_b(field_185512_D) && !p_196271_1_.func_196955_c(p_196271_4_, p_196271_5_) ? Blocks.field_150350_a.func_176223_P() : super.func_196271_a(p_196271_1_, p_196271_2_, p_196271_3_, p_196271_4_, p_196271_5_, p_196271_6_);
    }
 
-   public boolean func_176473_a(IBlockReader p_176473_1_, BlockPos p_176473_2_, BlockState p_176473_3_, boolean p_176473_4_) {
+   public boolean canGrow(IBlockReader p_176473_1_, BlockPos p_176473_2_, BlockState p_176473_3_, boolean p_176473_4_) {
       return p_176473_3_.func_177229_b(field_176501_a) < 2;
    }
 
@@ -90,10 +91,14 @@
    }
 
    public void func_225535_a_(ServerWorld p_225535_1_, Random p_225535_2_, BlockPos p_225535_3_, BlockState p_225535_4_) {
-      p_225535_1_.func_180501_a(p_225535_3_, p_225535_4_.func_206870_a(field_176501_a, Integer.valueOf(p_225535_4_.func_177229_b(field_176501_a) + 1)), 2);
+      grow(p_225535_1_, p_225535_2_, p_225535_3_, p_225535_4_, (net.minecraftforge.plants.IFertilizer) net.minecraft.item.Items.field_196106_bc);
    }
 
    protected void func_206840_a(StateContainer.Builder<Block, BlockState> p_206840_1_) {
       p_206840_1_.func_206894_a(field_185512_D, field_176501_a);
    }
+   public void grow(ServerWorld world, Random rand, BlockPos pos, BlockState state, net.minecraftforge.plants.IFertilizer fertilizer) {
+       world.func_180501_a(pos, state.func_206870_a(field_176501_a, Math.min(2, Integer.valueOf((int) (state.func_177229_b(field_176501_a) + 1 * fertilizer.getPotency())))), 2);
+   }
+    @Override public BlockState getPlant(IBlockReader world, BlockPos pos) { return world.func_180495_p(pos); }
 }
